### ISSUE TITLE
Train and Embedding from Speaker Recognition and Verification

### ISSUE THREAD

**Issue by Iagodel** [NVIDIA/NeMo#2713]
Hello guys, I have two question.
First, I had a "small" dataset, around 20 hours for some tests.  I successfully trained this dataset but I had some doubts about this train, if I train correctly or not. So, do you give an advice about the train in general? Like how much hours I need to train, an order to do the train, any recommendation if possible.

Second, I tested some audios with voxceleb_eval.py in two moments, with a unique embedding and with two different embedding (that sounds a good ideia for a test),  and I got good results, but I need to do it if audios that won't be in train or test dataset. Like a test in transcritor, I trained a large dataset but when I will test the train, I put an audio that won't be in the train. Is that possible? What I need to do?

**Comment by nithinraok** [comment]
1. There is no general rule of thumb for how long you can train. The best thing is to separate a part of the train to dev and make sure both of them go low as much as possible. For voxceleb with 7205 classes (~1900hrs), we train for 200 epochs with a batch size of 64, SGD optimizer, and cosine annealing as scheduler. So I would recommend you to start with 30 epochs with experimenting on learning rate and weight decay. Then you will get an idea of how the network training and the best possible hyperparameters. 
2. I am not clear on your question, could you elaborate well? If you are looking to get embeddings for a set that's not part of your train and test but has similar domain data on which model was trained, look at embedding_extraction [script](https://github.com/NVIDIA/NeMo/blob/main/examples/speaker_recognition/extract_speaker_embeddings.py) , this will extract the embeddings for audio samples and store embeddings in the current `./data/` folder in pickle format

**Comment by Iagodel** [comment]
1. Okay I got it the ideia, thanks a lot @nithinraok. I will try on this way.

2. Of course, I will try again. Imagine, you have a small dataset, something like 100 hrs for train and 5 hrs to dev (this is only to examplify). But you need to test another audio that wouldn't be on the train set or dev set, even this voice being or not on dataset. That's mean, you are testing your model with a voice that it know, but it isn't on the train or test set. More clearly now?

**Comment by nithinraok** [comment]
For getting embeddings on unknown set you have two options:
1. If there is a single audio file, you can just use this single liner to get embeddings:
`speaker_model = ExtractSpeakerEmbeddingsModel.from_pretrained(model_name="speakerverification_speakernet")
embs = speaker_model.get_embedding('audio_path')`
2. If there are more than a single audio file, you can just use embedding_extraction [script](https://github.com/NVIDIA/NeMo/blob/main/examples/speaker_recognition/extract_speaker_embeddings.py), this will extract the embeddings for audio samples and store embeddings in the current `./embeddings/` folder in pickle format
